src/app/3d/qgsgoochmaterialwidget.cpp

```diff
@@ -71,5 +80,13 @@ QgsAbstractMaterialSettings *QgsGoochMaterialWidget::settings()
   m->setShininess( spinShininess->value() );
   m->setAlpha( spinAlpha->value() );
   m->setBeta( spinBeta->value() );
+
+  QgsPropertyCollection dataDefinedPropertie;
```

Comment:
Here you could use

    const QList< QgsPropertyOverrideButton * > buttons = findChildren< QgsPropertyOverrideButton * >();

and loop over the found widgets to set their properties. That avoids more duplicate code.